### PR TITLE
refactor: update moderation request

### PR DESCRIPTION
**Pull request by josenriagu**
## Description

- sets correct reason on entry update after reporting

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: apps/moderation/src/extensions/report-modal.tsx [hunk 1]
```diff
@@ -57,27 +57,27 @@ const ReportModalComponent = (props: RootComponentProps) => {
     }
   }, [activeModal]);
 
-  const updateOnSuccess = (isSuccess: boolean) => {
+  const updateOnSuccess = (isSuccess: boolean, reason: string) => {
     // this method utilises react-query to update reported state depending on contentType
     if (contentType === 'post') {
       queryClient.setQueryData<unknown>([ENTRY_KEY, activeModal.entryId], prev => ({
```
Comment (SeverS) [inline apps/moderation/src/extensions/report-modal.tsx]: This needs a refactor. I am not ok updating the state outside of a hook. it makes it untraceable.
I would wrap the call that is made through the `moderationRequest.clickHandler` with a `useMutation` and do the mutation directly. This way you will also benefit from the status and error

File: ui/hooks/src/moderation-request.ts [hunk 2]
```diff
@@ -255,7 +255,7 @@ export default {
             throw new Error('Unable to process your request right now. Please try again later');
           }
 
-          return modalName === 'report-modal' ? callback(true) : callback();
+          return modalName === 'report-modal' ? callback(true, dataToSign.reason) : callback();
```
Comment (SeverS) [inline ui/hooks/src/moderation-request.ts]: I would suggest to always pass those parameters to the callback and get rid of this logic here, and remove the unnecessary modalName param.

File: ui/hooks/src/moderation-request.ts [hunk 3]
```diff
@@ -255,7 +255,7 @@ export default {
...
         })
         .catch(error =>
           logger.error(`[${modalName}.tsx]: fetchRequest err %j`, error.message || ''),
```
Comment (SeverS) [inline ui/hooks/src/moderation-request.ts]: when you log the error don't lie about the source. this is not the `${modalName}.tsx` file. It's the moderation-request.modalClickHandler. also make sure that you pass only one parameter to the `.error` and it is a `string`

**Review by SeverS — CHANGES_REQUESTED** [review]

File: ui/hooks/src/moderation-request.ts (hunk 3 above)
Comment (josenriagu) [reply ui/hooks/src/moderation-request.ts]: gotcha.

**Review by SeverS — APPROVED** [review]

**Pull request merged**